docs/source/app_development/widgets.rst

```diff
@@ -1,106 +1,91 @@
 
-In ipywidgets, all the widgets are using the `traitlets <https://traitlets.readthedocs.io/>`_ types for the parameters.
-As shown in the example above, the value of the ``FloatSlider`` is a ``Float`` trait type.
-One can not only show the current value ``value["new"]`` but also the old value ``value["old"]``.
-Traits can emit change events when the attributes are updated.
-This is very useful for creating custom widgets.
 
-Custom widgets
-==============
+**************************
+Terminology and background
+**************************
 
-Ipywidgets package also supports developers in creating their own widgets.
-There is a detailed `tutorial <https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Custom.html>`_ to show how to develop a custom widget.
-In the ipywidgets framework is based on the Comm framework.
-In the Comm framework, the change events of the traits leads to send a json massage to the Jupyter kernel.
-Then, the frontend will update accordingly.
+Widgets and traitlets
+======================
 
-Develop the custom widgets mainly involve two parts:
+Widgets are eventful Python objects that display GUI components, such as a sliders, buttons, progress bars, drowdowns, etc.
+`ipywidgets`_ is a Python package that provides interactive widgets for the use in Jupyter notebooks.
 
-* **Python**: in the Python part, one needs to define the APIs of the widgets.
-  For instance, the color of the widget's background.
-  The traits types should be initialized as:
+An Ipywidget has a parameter, whose value can be ovserved and accessed from Python.
+This is implemented using `traitlets <https://traitlets.readthedocs.io/>`_.
 
+For example, try the following code in a Jupyter notebook:
 
 .. code-block:: python
 
-    from traitlets import Unicode
-
-    color = Unicode('').tag(sync=True)
-
-* **Javascript** (or TypeScript): the javascript part is for the frontend of the widgets.
-  The frontend uses the `backbone.js <https://backbonejs.org/>`_ framework.
-  The communications between the Python and Javascript parts are from the set and get functions of the traits.
-
-.. code-block:: javascript
+    from ipywidgets import FloatSlider
 
-    let color = this.model.get('color');
+    # slider widget that has a `value` traitlet attribute
+    slider = FloatSlider(value=273.0, min=100.0, max=500.0, description="Temperature [K]");
 
-    this.model.set('color', 'red');
+    # Access the slider value as `slider.value`
+    # print(slider.value)
 
-The change of the traits can be monitored and triggle javascript function.
+    def slider_change(change):
+        """Handle slider changes.
 
-.. code-block:: javascript
+        This function is called when the slider value is changed by the user.
+        """
+        print(change["old"])
+        print(change["new"])
+        if change["new"] > 373.0:
+            print("Boiling now!")
 
-    this.model.on('change:color', this._color_change);
+    slider.observe(slider_change, names="value")
```

Comment:
I am usually not a fan to maintain documentation for other packages here, since APIs and best practices may change. If we do so, I'd prefer with as little code as possible to make it a true "minimal example", I believe this could be achieved by removing the print function calls in lines 72 and 73.

Reply:
I generally agree, which is why I removed the more involved python+JS example and only left this one in.

Will also remove lines 72+73